### DISCUSSION TITLE
Adding additional generic to WebSocketSubject

### DISCUSSION THREAD

**Discussion by jkossis** [ReactiveX/rxjs#6018] [Ideas / Feature request]
Both `WebSocketSubject<T>` and it's creator function `webSocket<T>` currently only take one generic, for distinguishing the type that the internal Subject holds. As it currently stands, `.next(data: T)` and `.subscribe((data: T) => { ... })` force the data to be of the same type.

However, data emitted to the web socket server could be of a different shape/form than the data received from the web socket server. What are people's thoughts on extending this functionality to include a 2nd generic, perhaps of type `R`, to allow for stronger typing on non-symmetrical communication?

**Comment by cartant** [comment]
Related #5381 and #5936.
